### DISCUSSION TITLE
Compile issues on Windows with v1.0.0

### DISCUSSION THREAD

**Discussion by annnei** [duckdb/duckdb#12487] [Q&A]
I am trying to compile duckdb on Windows using MSVC, and that has been working fine historically, but today I moved over to v1.0.0 and now I get:

```
...\duckdb-src\third_party\libpg_query\postgres_parser.cpp(43): error C2243: 'type cast': conversion from 'std::vector<duckdb_libpgquery::PGKeyword,std::allocator<duckdb_libpgquery::PGKeyword>> *' to '_Ty &&' exists, but is inaccessible
          with
          [
              _Ty=duckdb::vector<duckdb_libpgquery::PGKeyword,true>
          ]

```

Is this some known issue? Any ideas about a workaround?

**Comment by carlopi** [comment]
This is not great, but I am not sure what's wrong here, given we compile fine on a wide hosts of compilers (also MSVC).

Fix I think is patching third_party\libpg_query\postgres_parser.cpp from:
```
vector<duckdb_libpgquery::PGKeyword> PostgresParser::KeywordList() {
        // FIXME: because of this, we might need to change the libpg_query library to use duckdb::vector
        return std::forward<vector<duckdb_libpgquery::PGKeyword> >(duckdb_libpgquery::keyword_list());
}
```
to possibly:
```
vector<duckdb_libpgquery::PGKeyword> PostgresParser::KeywordList() {
        // FIXME: because of this, we might need to change the libpg_query library to use duckdb::vector
        vector<duckdb_libpgquery::PGKeyword> tmp (duckdb_libpgquery::keyword_list());
        return std::move(tmp);
}
```
or something similar.

If you send a PR it would be amazing.
Double amazing if you can help reproduce your setup in a dockerized version or within CI, so that it can avoided going forward.

**Comment by annnei** [comment]
Sorry for the delay, I can confirm that the following code compiles in my MSVC setup:

vector<duckdb_libpgquery::PGKeyword> PostgresParser::KeywordList() {
      // FIXME: because of this, we might need to change the libpg_query library to use duckdb::vector
      vector<duckdb_libpgquery::PGKeyword> tmp(duckdb_libpgquery::keyword_list());
      return tmp;
}

I.e. MSVC issued a warning against std::move on a temporary variable, but the code above works.

Unfortunately I don't have the environment to reproduce it in a dockerized version at present, but I'll try to create a PR within the next few days, unless you beat me to it.

Thanks for the assistance.

Regards,
Anna Neiderud
________________________________

**Comment by annnei** [comment]
PR posted: https://github.com/duckdb/duckdb/pull/12579

**Comment by carlopi** [comment]
Thanks, I will take a deeper look and leave a review, either your solution on an alternative one will then land.
You/whoever search for this also have a way forward, also great.
